pkg/serviceaccount/claims.go

```diff
@@ -120,7 +121,7 @@ func (v *validator) Validate(_ string, public *jwt.Claims, privateObj interface{
 		klog.V(4).Infof("Could not retrieve service account %s/%s: %v", namespace, saref.Name, err)
 		return nil, err
 	}
-	if serviceAccount.DeletionTimestamp != nil {
+	if serviceAccount.DeletionTimestamp != nil && !serviceAccount.DeletionTimestamp.Time.After(nowTime) {
```

Comment:
What about the grace period? If a pod is gracefully shutting down, should it still be able to authenticate?

Reply:
the deletionTimestamp reflects the grace period... if I delete at time T with a grace period of 30 seconds, the deletionTimestamp is set to T+30